x-pack/legacy/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -350,6 +370,43 @@ export class VectorLayer extends AbstractLayer {
     };
   }
 
+  async _performInnerJoins(sourceResult, joinStates, updateSourceData) {
+
+    let shouldUpdate = false;
+    for (let i = 0; i < sourceResult.featureCollection.features.length; i++) {
+      const feature = sourceResult.featureCollection.features[i];
+      const oldVisbility = feature.properties[FEATURE_VISIBLE_PROPERTY_NAME];
+      feature.properties[FEATURE_VISIBLE_PROPERTY_NAME] = true;//new visibility will be determined based on join-results
```

Comment:
Don't need this line. `feature.properties[FEATURE_VISIBLE_PROPERTY_NAME]` is just getting set to canJoin below.

Reply:
see other comment as to why this was needed